packages/my-near-wallet/src/lib/my-near-wallet.ts

```diff
@@ -75,26 +76,36 @@ const MyNearWallet: WalletBehaviourFactory<
   { params: MyNearWalletExtraOptions }
 > = async ({ metadata, options, store, params, logger }) => {
   const _state = await setupWalletState(params, options.network);
-
   const getAccounts = async (): Promise<Array<Account>> => {
     const accountId = _state.wallet.getAccountId();
     const account = _state.wallet.account();
 
     if (!accountId || !account) {
       return [];
     }
-
-    return [
-      {
-        accountId,
-        publicKey: (
-          await account.connection.signer.getPublicKey(
-            account.accountId,
-            options.network.networkId
-          )
-        ).toString(),
-      },
-    ];
+    const publicKeyTemp = await account.connection.signer.getPublicKey(
+      account.accountId,
+      options.network.networkId
+    );
```

Comment:
It looks like `signIn` without a `contractId` is still not working for **MyNearWallet** the error is not showing but you can not sign in either:

I would suggest that we change the `getAccounts` function into something like this:

```ts
const getAccounts = async (): Promise<Array<Account>> => {
    const accountId = _state.wallet.getAccountId();
    const account = _state.wallet.account();
    if (!accountId || !account) {
      return [];
    }
    try {
      const publicKey = await account.connection.signer.getPublicKey(
        account.accountId,
        options.network.networkId      );
      return [
        {
          accountId,
          publicKey: publicKey.toString(),
        },
      ];
    } catch (e) {
      return [
        {
          accountId,
          publicKey: undefined,
        },
      ];
    }
  };
```

In the past, the `getAccounts` method was not `async` so the best we can do for now is check and catch any error while trying to get the `publicKey` when there are no errors we return the values for `accountId` and `publicKey` if there are errors for `publicKey` we return `undefined`.

Reply:
Thank you for info and your example of code 👍